### ISSUE TITLE
Text color fails when textbox rotated

### ISSUE THREAD

**Issue by shaehn** [chunyenHuang/hummusRecipe#80]
I am finding an odd behavior when I colorize text in a textbox and rotate it.  When I use a primary color, say red (0xFF0000), the rotated text will be as expected, red. However, when I use an alternate color flavoring, say 0x974435, the rotated text becomes white. So what is going on?

Also, I think I found a problem when supplying a three element array for the RGB colors.  The code in text.js is expecting an object looking like {r:0,g:0,b} and the vector_helper.js function is simply sending back the array. So the following code in vector_helper.js

```javascript
function transformColorToRGB(code) {
    return (Array.isArray(code)) ? code : hexToRgb(code);
}
```
should be:

```javascript
function transformColorToRGB(code) {
    return (Array.isArray(code)) ? { r:code[0], g:code[1], b:code[2] } : hexToRgb(code);
}
```

**Comment by shaehn** [comment]
OK,  I figured out what is going on. The problem is in the text.js file where the text rotation object is found.

The original code is as follows:
```
 xObject
                    .getContentContext()
                    .q()
                    .gs(xObject.getGsName(options.fillGsId))
                    .BT()
                    .rg(color.r, color.g, color.b)
                    .Tf(options.font, options.size)
                    .Tm(1, 0, 0, 1, 1, 1)
                    .Tj(text)
                    .ET()
                    .Q();
                xObject.end();
```
The problem lies with the arguments to the .rg function. They are supposed to be fractional, but they are not. Each needs to be divided by 255 to get the correct number.  Thus the code would be:

```
 xObject
                    .getContentContext()
                    .q()
                    .gs(xObject.getGsName(options.fillGsId))
                    .BT()
                    .rg(color.r/255, color.g/255, color.b/255)
                    .Tf(options.font, options.size)
                    .Tm(1, 0, 0, 1, 1, 1)
                    .Tj(text)
                    .ET()
                    .Q();
                xObject.end();
``
Also, the other change to vector_helper.js as noted above would also need to be done,.

**Comment by shaehn** [comment]
I would be willing to put this fix in place (I would choose just modifying vector_helper.js), but I would need some guidance on how to go about it to make sure the pull request get absorbed relatively quickly because of my local need. Let me know, thanks.

**Comment by chunyenHuang** [comment]
PR please! I like the approach to modify the `hexToRgb` function

**Comment by shaehn** [comment]
OK, I can do that. In the meantime, I found another problem related to the rotation of the text. When the fill color is provided, the fill rectangle has not mechanism to rotate, so it just sits there in 0 degree position while the text rotates out of the box. I have a fix for that also involving text.js, vector.helper.js, and vector.js. Would it be alright to put theses fixes under the one PR, or do you want another issue generated?

**Comment by chunyenHuang** [comment]
2 PRs will be great, so we can refer the issues individually. Thanks for you help

**Comment by shaehn** [comment]
How does the documentation get updated?  The rotation nomenclature for text indicates rotation is fixed to 0, 90, 189,... and that is no longer true? Same goes for the rectangle fix I am going to put in. Rectangles will now be able to be rotated.

**Comment by chunyenHuang** [comment]
Fix in `hummus-recipe@1.8.2`
